graphics/scaler/hq.cpp

```diff
@@ -4987,72 +4947,90 @@ static void HQ3x_implementation(const uint8 *srcPtr, uint32 srcPitch, uint8 *dst
 	}
 }
 
-HQScaler::HQScaler(const Graphics::PixelFormat &format) : Scaler(format) {
+HQScaler::HQScaler(const Graphics::PixelFormat &format) : Scaler(format),
+#ifdef USE_NASM
+	_hqx_params(nullptr),
+#endif
+	_RGBtoYUV(nullptr) {
 	_factor = 2;
 
 	if (format.bytesPerPixel == 2) {
-		InitLUT(format);
+		initLUT(format);
 	} else {
 		// Pass a 16 bit 565 format to InitLut
 		Graphics::PixelFormat format16(2,
 		                               5, 6, 5, 0,
 		                               11, 5, 0, 0);
-		InitLUT(format16);
+		initLUT(format16);
 	}
 }
 
 HQScaler::~HQScaler() {
-	free(RGBtoYUV);
-	RGBtoYUV = 0;
+	delete _RGBtoYUV;
```

Comment:
I think this should be delete []?

It almost never makes a difference for basic types but nice to be correct.

Reply:
Good catch!
Thank you very much.